cli/azd/pkg/templates/template_manager.go

```diff
@@ -43,3 +50,141 @@ func (tm *TemplateManager) GetTemplate(templateName string) (Template, error) {
 func NewTemplateManager() *TemplateManager {
 	return &TemplateManager{}
 }
+
+type gitRepository struct {
+	host   string
+	slug   string
+	branch string
+}
+
+func (repoInfo *gitRepository) downloadZipUrl() string {
+	return fmt.Sprintf(
+		"https://%s/%s/archive/%s.zip",
```

Comment:
I believe this will only work for GitHub. We'll need this to work for any git repo. AzDO gitlab, etc.

Reply:
> I believe this will only work for GitHub. We'll need this to work for any git repo. AzDO gitlab, etc.

I do not want us to invest a lot of effort in trying to make this work for every possible forge out there.  If AzDo and GitLab support getting a zip file of the code as GitHub does, it's great and we can use that.

I think if folks want to use templates from other forges it's reasonable to expect them to have `git` installed.  I don't want us to try to implement pulling down code from a git server exposed over HTTP in `azd` itself.